src/ERC1155D.sol

```diff
@@ -224,10 +221,15 @@ contract ERC1155D is ERC165, IERC1155, IERC1155MetadataURI {
      */
 
     function _mintSingle(address to, uint256 id) internal virtual {
-        require(to != address(0), "ERC1155: mint to the zero address"); // you can remove this if only minting to msg.sender
-        require(ownerOf[id] == address(0), "ERC1155D: supply exceeded");
+        // can be removed since we mint only when ownerOf[id] is always 0.
+        // require(ownerOf[id] == address(0), "ERC1155D: supply exceeded");
+
+        ownerOf[id] = to;
 
-        ownerOf[id] = to; // this can be made more efficient with assembly if you know what you are doing!
+        if (to.code.length != 0) {
+            require(IERC1155Receiver(to).onERC1155Received(msg.sender, address(0), id, 1, "")
```

Comment:
I'm not we should make a mint call `onERC1155Received`. It's technically not a transfer, and so the call is not mandated by the 1155 spec. I'm open adding this to encourage composability, but if we do the `_mintSingle` call in `duplicate` should be moved after the ERC20 token transfer

Reply:
> It's technically not a transfer, and so the call is not mandated by the 1155 spec

Doesn't seem like the case to me: See "onERC1155Received rules:" here https://eips.ethereum.org/EIPS/eip-1155#safe-transfer-rules

Reply:
Ah yes, i see now. I swa the first part of this:
```
A contract MAY skip calling the ERC1155TokenReceiver hook function(s) if the mint operation is transferring the token(s) to itself. In all other cases the ERC1155TokenReceiver rules MUST be followed as appropriate for the implementation (i.e. safe, custom and/or hybrid).
```
but missed the second half.

I still think that the `_mintSingle` call should be moved to either directly before or directly after the ERC20 token transfer.

Reply:
done.